apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/Indictment.tsx

```diff
@@ -72,20 +76,78 @@ const Indictment: React.FC = () => {
 
   useDeb(workingCase, ['indictmentIntroduction', 'demands'])
 
+  const setDriversLicenseSuspensionRequest = useCallback(
+    (indictmentCounts?: TIndictmentCount[]) => {
+      let requestDriversLicenseSuspension = false
+
+      const reduce = () =>
+        indictmentCounts?.forEach((count) =>
+          count.offenses?.forEach((offence) => {
+            switch (offence) {
+              case IndictmentCountOffense.DrivingWithoutLicence:
+                requestDriversLicenseSuspension = false
+                return
+              case IndictmentCountOffense.DrunkDriving:
+                if (
+                  count.substances &&
+                  count.substances[IndictmentCountOffense.DrunkDriving] &&
+                  count.substances[IndictmentCountOffense.DrunkDriving][
+                    'ALCOHOL'
+                  ] >= '1,20'
+                ) {
+                  requestDriversLicenseSuspension = true
+                }
+                break
+              case IndictmentCountOffense.IllegalDrugsDriving:
+              case IndictmentCountOffense.PrescriptionDrugsDriving:
+                requestDriversLicenseSuspension = true
+                break
+              default:
+                break
+            }
+          }),
+        )
+
+      reduce()
+
+      if (
+        requestDriversLicenseSuspension !==
+        workingCase.requestDriversLicenseSuspension
+      ) {
+        setAndSendCaseToServer(
+          [{ requestDriversLicenseSuspension, force: true }],
+          workingCase,
+          setWorkingCase,
+        )
+      }
+    },
+    [setAndSendCaseToServer, setWorkingCase, workingCase],
+  )
+
   const handleCreateIndictmentCount = useCallback(async () => {
     const indictmentCount = await createIndictmentCount(workingCase.id)
 
     if (!indictmentCount) {
       return
     }
 
+    const indictmentCounts = workingCase.indictmentCounts
+      ? [...workingCase.indictmentCounts, indictmentCount]
+      : [indictmentCount]
```

Comment:
```suggestion
    const indictmentCounts = [...(workingCase.indictmentCounts || []), indictmentCount]
```

Reply:
I think this way is a little more readable, what do you think?

Reply:
We should use `??` rather than logical or in this case. I will make the change.